packages/waku/src/lib/plugins/vite-plugin-typegen.ts

```diff
@@ -0,0 +1,102 @@
+import type { Plugin } from 'vite';
+import { promises } from 'fs';
+import path from 'path';
+
+const SRC_DIR = path.resolve(process.cwd(), './src/pages');
```

Comment:
seems hard coded. and we don't use `cwd`.

let's revisit when this PR is ready.

Reply:
time to revisit I think haha. Any suggestions for how to make this less hard-coded?

Reply:
Follow up to this is: How should we detect when a user is using fs-router or not? This also seems like something that might suggest some need to hardcode, but I'm curious if you know a better way.

Reply:
- Use `config.root` instead of `process.cwd`.
- `srcDir` is in config, so get it through plugin options. See other plugins.
- `'pages'` must be hard-coded for now.
- Related, we can't detect if fsRouter is used, but if we don't find `entries.tsx` (aka "managed mode"), then we run the plugin. (see rscManagedPlugin)

Reply:
all updated 👍